app/src/main/java/co/develhope/chooseyouowncocktail_g2/DetailDrinkPageFragment.kt

```diff
@@ -44,34 +46,30 @@ class DetailDrinkFragment : Fragment() {
         return binding.root
     }
 
+
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        binding.title.text = param_name
-        binding.description.text = param_desc
-
-        Picasso.get()
-            .load("https://www.thecocktaildb.com/images/media/drink/metwgh1606770327.jpg")
-            .resize(200,200)
-            .placeholder(R.drawable.placeholder)
-            .error(R.drawable.placeholder)
-            .into(binding.preview)
-
-        binding.cl.text = param_cl
-
-        requireActivity().onBackPressedDispatcher.addCallback(viewLifecycleOwner) {
-            when (param_currentPage) {
-                "Search" -> view.findNavController()
-                    .navigate(R.id.action_detailDrinkFragment_to_navigation_search)
-                "Home" -> view.findNavController()
-                    .navigate(R.id.action_detailDrinkFragment_to_navigation_home)
+        arguments?.let {
+            DrinkList.getByID(it.getInt(param_drink_ID)).let {
+                binding.title.text = it!!.name
+                binding.description.text = it.description
+                binding.cl.text = it.cl.toString()
+                binding.preview.setImageByUrl(
+                    "https://www.thecocktaildb.com/images/media/drink/metwgh1606770327.jpg",
+                    200,
+                    200
+                )
             }
         }
 
-        var switch:Boolean= false
-        var bf=binding.buttonFavorite
-        val imOn= requireActivity().getResources().getIdentifier("ic_fav_off","drawable", requireContext().getPackageName())
-        val imOff= requireActivity().getResources().getIdentifier("ic_fav_on","drawable", requireContext().getPackageName())
+
+        var switch: Boolean = false
+        var bf = binding.buttonFavorite
+        val imOn = requireActivity().getResources()
```

Comment:
Non è il modo più corretto di accedere ad una risorsa e di settarla

Reply:
Si, ma come ti ho già scritto nei commenti precedenti appartiene ad un branch già mergiato su develop, non a questo e il problema è stato già preso in carico in un altro ticket

Reply:
Si, quella di @FraPedu1 , se non sbaglio